Fix Hindi locale inconsistencies for chapters 1–5

aniket866 opened the pull request:
# Fixes and Improvements of Hindi locale inconsistencies and i18n/config.ts(Chapters 1-5):

- [x] Build was run locally and any changes were pushed
- [x] Title clearly describes the changes
- [x] Issue referenced in the PR description (not in commits or title)
- [x] Clean commit history

## Overview
This PR addresses multiple inconsistencies identified during the recent i18n audit, with a specific focus on fixing Hindi locale inconsistencies across Chapters 1 to 5 and updates to i18n/config.ts  The primary scope includes standardizing locale definitions, correcting improper Hindi locale usage, and ensuring consistent i18n behavior.

## 1. Dead / Broken Internal Lesson Links
* **Issue:** Internal routes like `public-key-3/help` did not exist, leading to 404 errors.
* **Fix:** Replaced broken internal links with context-aware external **ChatBTC (bitcoinsearch)** links to provide conceptual and technical explanations.

-   Adjusted tooltip positioning to prevent UI overlap.

    * Integrated ChatBTC search links inside tooltips.
    * The tool tip was not embedded under a valid link and also It was not defined , it was also overlapping with text there so I corrected this as a part of fixing Hindi Locale inconsistencies, look at the tool tips: "block explorer" and "Hex"**
<img width="1366" height="768" alt="image" src="https://github.com/user-attachments/assets/84862aa9-950c-4af9-9fb0-27ce47f86599" />
<img width="1366" height="768" alt="image" src="https://github.com/user-attachments/assets/0bf6a7ca-7604-4dcf-b93f-ed70a53778c9" />

## 3. Duplicate English Content on Hindi Pages
* **Issue:** Selected Hindi pages displayed Hindi text followed by duplicated English blocks.
* **Root Cause:** Incorrect paragraph mapping and missing separation in translation files.
* ** Current (Incorrect Behavior)

When Hindi was selected, the story content appeared in **both Hindi and English**:

This caused mixed-language content on a Hindi-only page.
<img width="1366" height="768" alt="image" src="https://github.com/user-attachments/assets/64a51113-280c-417e-bde9-b7e2b134c91c" />

- Removed the duplicated English content from the Hindi locale
- Ensured that only Hindi text is rendered when the Hindi language is selected


## 3. Missing Paragraphs in Translations
* **Issue:** Hindi sections were missing entire paragraphs present in the English source on Chapter 5.
<img width="1356" height="561" alt="image" src="https://github.com/user-attachments/assets/5c5234b2-1717-466d-ba51-640fa202808b" />

* **Result:** Achieved 100% content parity between English and Hindi.

 @satsie  Please review this , this can be merged , I am next going to debug from chapter 6 to 10 and I will raise another PR for that.

aniket866 committed:
fix(i18n): fix dead links in Hindi and Spanish chapter four

aniket866 committed:
Fix Hindi locale inconsistencies for chapters 1-5

0tuedon commented on i18n/config.ts:
```diff
@@ -11,7 +11,7 @@ export const i18n = {
     { locale: 'ko', label: '한국어' },
     { locale: 'ja', label: '日本語' },
-    { locale: 'zh_cn', label: '中文' }
+    { locale: 'zh-CN', label: '中文' },
```
This shouldn't be changed

aniket866 replied:
I reverted the change ,you asked.

aniket866 committed:
Change locale key from 'zh-CN' to 'zh_cn'

satsie reviewed (approved):
tACK. I made sure I could still switch between languages, and checked out a few of the pages that were updated. Thanks for working on this!

Will wait for Tuedon to have a second look before merging.

satsie reviewed (changes requested):
@aniket866 I'm requesting changes regarding the PR description. I know it seems trivial but part of making meaningful open-source contributions is getting this part right. 

Please make sure your PR descriptions are concise. There are a number of things in this description that make the PR unnecessarily complicated to review. The harder you make it for reviewers, the longer it takes for your PR to be looked at and, hopefully, merged. Here are some examples of what I am talking about:

> 1. Locale / Intl Runtime Error
> Issue: A RangeError: Incorrect locale information provided was causing runtime crashes.
> Root Cause: An invalid locale identifier (zh_cn) was defined in the i18n configuration. JavaScript Intl APIs require valid BCP-47 locale tags.
> Fix: Replaced zh_cn with zh-CN. No other locale logic was modified.

For a runtime error like this you should provide some logs or a screenshot of where and when it occurs. 

After looking into BCP-47 more I agree that `zh-CN` is the standard, but changing it to that would go against the naming convention used for all the other locales. If you change one you would have to change the rest of them. You would also need to QA this and update the guidelines for new translation contributions. While it may seem like a small change, it's larger than it looks, and unrelated to the mostly Hindi translations in the rest of the change set. It's better to have something like this in a separate PR.

> Fixes: * Adjusted tooltip positioning to prevent UI overlap.

Can you identify where this is in the changeset? It wouldn't be in a translation file, and there are no other files changed in this PR.

> Replaced some tooltips with underlined external links for better UX.

Did you only do this for the Hindi translation? Can you provide an example?

> 6. Genesis Chapter Tooltip & Content Fixes
> Fix: Corrected tooltip usage in the Genesis chapter and ensured stable external references.
> Result: Aligned Hindi implementation with the Hindi reference.

What does this mean? How are the changes to the genesis chapter different from the changes you made to the other chapters?

aniket866 commented:
@satsie I really apologize for the inconvenience and will take care of this further.

Replaced zh_cn with zh-CN........
**This change was actually I made due to getting a runtime error while correcting Hindi Locale. by correcting an invalid locale tag (zh_cn → zh-CN)
but @0tuedon later said that must not be changed so I undone this change**

Adjusted tooltip positioning to prevent UI overlap........
**During Hindi translation the tool tip was not embedded under a valid link and also It was not defined , it was also overlapping with text there so I corrected this as a part of fixing Hindi Locale inconsistencies, look at the tool tips: "block explorer" and "Hex"**
<img width="1366" height="768" alt="image" src="https://github.com/user-attachments/assets/84862aa9-950c-4af9-9fb0-27ce47f86599" />
<img width="1366" height="768" alt="image" src="https://github.com/user-attachments/assets/0bf6a7ca-7604-4dcf-b93f-ed70a53778c9" />

Aligned Hindi implementation with the Hindi reference.........
**sorry for this one this should be "Aligned Hindi implementation with the English references" this actually mean a tool tip that was originally English worded is now also translated to Hindi for example a tool tip actually saying "what is block explorer" is also converted to Hindi "ब्लॉक एक्स्प्लोरर क्या है?"**

Once again, I apologize for the inconvenience caused. I will take care of every detail and provide references for each change.

aniket866 committed:
Merge branch 'dev' into fix/locale-hindi-chapter-1-to-5

satsie commented:
@aniket866

> Replaced zh_cn with zh-CN........
> This change was actually I made due to getting a runtime error while correcting Hindi Locale. by correcting an invalid locale tag (zh_cn → zh-CN)
> but @0tuedon later said that must not be changed so I undone this change

All this is fine, but can you please remove it from the PR description? It's not in the final code that you are asking to have reviewed.

> Adjusted tooltip positioning to prevent UI overlap........
> During Hindi translation the tool tip was not embedded under a valid link and also It was not defined , it was also overlapping with text there so I corrected this as a part of fixing Hindi Locale inconsistencies, look at the tool tips: "block explorer" and "Hex"

Wow! Was this only for the Hindi translation? Good catch. Can you please update the PR description with this bug description and the screenshots?

>  Aligned Hindi implementation with the Hindi reference.........
> sorry for this one this should be "Aligned Hindi implementation with the English references" this actually mean a tool tip that was originally English worded is now also translated to Hindi for example a tool tip actually saying "what is block explorer" is also converted to Hindi "ब्लॉक एक्स्प्लोरर क्या है?"

Same story here, please update this part of the PR description to reflect this clarification.

I also want to follow up on this one:

> Replaced some tooltips with underlined external links for better UX.

Is this referring to some of the tool tips not being defined? i.e. genesis_two.tooltip_block_explorer? If so, can you specify this in this part of the the PR description?

If and when this gets merged, someone is going to go through the release notes and view the changelog. The PR description should accurately reflect the changes that are being made. No one should have to read through the entire comment history of a PR to get a high level overview of the changes.

aniket866 commented:
Hi @satsie  I have updated the PR description and attached relevant screenshots for reference , Please have a look.

satsie reviewed (approved):
Thanks for cleaning up the PR description. You know what they say, a picture is worth a thousand words! The screenshots made it much easier to understand the changes that were made. 

There are still a few issues (i.e. there are two item number "3." sections, I'm pretty sure "Adjusted tooltip positioning to prevent UI overlap." should be item number "2.", and the commit history is not clean) but it's come a long way. Merging!
